src/OrchardCore/OrchardCore.Abstractions/OrchardCoreConstants.cs

```diff
@@ -28,4 +28,44 @@ public class Roles
 
         public const string Anonymous = "Anonymous";
     }
+
+    public class ConfigureOrder
+    {
+        public const int Default = 0;
+
+        // The configuration for authentication should be set up early, prior to any non-security modules.
+        public const int Authentication = -150;
+
+        // The reverse proxy should always be configured before the 'Authentication' and security initialization logic.
+        public const int ReverseProxy = Authentication * 2;
+
+        public const int Cors = ReverseProxy;
```

Comment:
CORS should be registered after the reverse proxy module to ensure the correct host is used.

Reply:
@kevinchalet I made this change. Any other suggestion or you good with this?